src/types.hpp

```diff
@@ -52,6 +52,9 @@ using ColorFloat = glm::vec4; // std::array<float, 4>;
 using MaterialId = uint8_t;
 using ColorId = uint8_t;
 
+static constexpr MaterialId MAT_ALL_MATERIAL = static_cast<MaterialId>(-1);
+static constexpr ColorId CLR_ALL_COLOR = static_cast<ColorId>(-1);
```

Comment:
```suggestion
static constexpr MaterialId MAT_NO_MATERIAL = static_cast<MaterialId>(-1);
static constexpr ColorId COLOR_NO_COLOR = static_cast<ColorId>(-1);
```

There's no such thing as an `ALL_MATERIAL`

Reply:
It was meant to be a * type thing. To capture all. I can change the name though.

For example imagine I have a function that returns a MaterialId. All tiles will return a valid MaterialId from a getter. Non-valid tiles will also return a valid MaterialId (0 representing air). Now if I return (-1) what does this mean? is it air? or am I saying use anything?

Reply:
Then name it `ANY_MATERIAL`